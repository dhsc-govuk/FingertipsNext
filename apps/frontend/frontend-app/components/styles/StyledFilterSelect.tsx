import styled from 'styled-components';
import { Select } from 'govuk-react';

export const StyledFilterSelect = styled(Select)({
  span: {
    fontWeight: 'bold',
  },
  select: {
    width: '100%',
  },
  marginBottom: '2em',
});
