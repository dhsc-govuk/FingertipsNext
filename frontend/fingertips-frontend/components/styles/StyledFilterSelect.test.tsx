import { render } from '@testing-library/react';
import { StyledFilterSelect } from './StyledFilterSelect';

describe('StyledFilterSelect', () => {
  it('should style the Select component as expected', () => {
    const container = render(
      <StyledFilterSelect label="Select an area">
        <option value="some value">some option</option>
      </StyledFilterSelect>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
