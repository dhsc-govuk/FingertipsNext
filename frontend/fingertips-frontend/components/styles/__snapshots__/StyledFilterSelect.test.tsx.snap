// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StyledFilterSelect > should style the Select component as expected 1`] = `
<DocumentFragment>
  <label
    class="src__Label-sc-iqzvxn-0 cjKMYe sc-aXZVg ciCqQj"
  >
    <span
      class="src__LabelText-sc-1lbxenh-0 bKqlAx"
    >
      Select an area
    </span>
    <select
      class="src__StyledSelect-sc-sgud4a-0 krelYQ"
    >
      <option
        value="some value"
      >
        some option
      </option>
    </select>
  </label>
</DocumentFragment>
`;

exports[`StyledFilterSelect should style the Select component as expected 1`] = `
<DocumentFragment>
  .c2 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 33px;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
}

.c3:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c1 {
  margin-bottom: 2em;
}

.c1 span {
  font-weight: bold;
}

.c1 select {
  width: 100%;
}

@media print {
  .c2 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c2 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    margin-bottom: 0;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

@media print {
  .c3 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    width: 50%;
    height: 38px;
  }
}

<label
    class="c0 c1"
  >
    <span
      class="c2"
    >
      Select an area
    </span>
    <select
      class="c3"
    >
      <option
        value="some value"
      >
        some option
      </option>
    </select>
  </label>
</DocumentFragment>
`;
